oidc-tokenstore/src/androidMain/kotlin/org/publicvalue/multiplatform/oidc/tokenstore/AndroidDataStoreSettingsStore.kt

```diff
@@ -0,0 +1,106 @@
+package org.publicvalue.multiplatform.oidc.tokenstore
+
+import android.content.Context
+import android.os.Build
+import android.security.keystore.KeyGenParameterSpec
+import android.security.keystore.KeyProperties
+import android.util.Base64
+import androidx.annotation.RequiresApi
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import java.security.KeyStore
+import javax.crypto.Cipher
+import javax.crypto.KeyGenerator
+import javax.crypto.SecretKey
+import javax.crypto.spec.GCMParameterSpec
+
+@RequiresApi(Build.VERSION_CODES.M)
+class AndroidDataStoreSettingsStore(
+    private val context: Context
+) : SettingsStore {
+
+    companion object {
+        private const val ANDROID_KEY_STORE = "AndroidKeyStore"
+        private const val KEY_ALIAS = "DataStoreEncryptionKey"
+        private const val TRANSFORMATION = "AES/GCM/NoPadding"
+        private const val IV_SIZE = 12
+        private const val TAG_SIZE = 128
+    }
+
+    // Use the by delegate for DataStore
+    private val Context.dataStore by preferencesDataStore(
+        name = "org.publicvalue.multiplatform.oidc.tokenstore"
+    )
+
+    // Get or generate the secret key
+    private val secretKey: SecretKey
+        get() {
+            val keyStore = KeyStore.getInstance(ANDROID_KEY_STORE).apply { load(null) }
```

Comment:
Maybe we can story the instance like in a lazy field?

Reply:
Gotcha 👍🏼